runtime/src/main/java/com/datastax/oss/quarkus/runtime/reactive/MutinyReactiveResultSet.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.quarkus.runtime.reactive;
+
+import com.datastax.dse.driver.api.core.cql.reactive.ReactiveResultSet;
+import com.datastax.dse.driver.api.core.cql.reactive.ReactiveRow;
+import com.datastax.oss.driver.api.core.cql.ColumnDefinitions;
+import com.datastax.oss.driver.api.core.cql.ExecutionInfo;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.groups.MultiBroadcast;
+import io.smallrye.mutiny.groups.MultiCollect;
+import io.smallrye.mutiny.groups.MultiConvert;
+import io.smallrye.mutiny.groups.MultiGroup;
+import io.smallrye.mutiny.groups.MultiOnCompletion;
+import io.smallrye.mutiny.groups.MultiOnEvent;
+import io.smallrye.mutiny.groups.MultiOnFailure;
+import io.smallrye.mutiny.groups.MultiOnItem;
+import io.smallrye.mutiny.groups.MultiOverflow;
+import io.smallrye.mutiny.groups.MultiSubscribe;
+import io.smallrye.mutiny.groups.MultiTransform;
+import java.util.concurrent.Executor;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+
+public class MutinyReactiveResultSet implements Multi<ReactiveRow>, ReactiveResultSet {
+
+  private final ReactiveResultSet reactiveResultSet;
+  private final Multi<ReactiveRow> multi;
+
+  public MutinyReactiveResultSet(ReactiveResultSet reactiveResultSet) {
+
+    this.reactiveResultSet = reactiveResultSet;
+    this.multi = Wrappers.toMulti(reactiveResultSet);
+  }
+
+  @NonNull
+  @Override
+  public Publisher<? extends ColumnDefinitions> getColumnDefinitions() {
+    return reactiveResultSet.getColumnDefinitions();
+  }
+
+  @NonNull
+  @Override
+  public Publisher<? extends ExecutionInfo> getExecutionInfos() {
+    return reactiveResultSet.getExecutionInfos();
+  }
+
+  @NonNull
+  @Override
+  public Publisher<Boolean> wasApplied() {
```

Comment:
Should return `Uni<Boolean>`.

Reply:
This is problematic because:
```
@NonNull
  @Override
  public Uni<? extends ColumnDefinitions> getColumnDefinitions() {
    Uni<? extends ColumnDefinitions> uni = Wrappers.toUni(reactiveResultSet.getColumnDefinitions());
  }
```
because it clashes with parent method and return incompatible type.
Should we create a 2nd specialized method (maybe on the `MutinyReactiveResultSet` interface) and implement it here?

Reply:
We can follow the same pattern as for other Uni, so:
```
@NonNull
  @Override
  public MutinyColumnDefinitions getColumnDefinitions() {
    
    return new MutinyColumnDefinitionsImpl(reactiveResultSet.getColumnDefinitions());
  }
```
then interface:
```
public interface MutinyColumnDefinitions extends Publisher<ColumnDefinitions>, Uni<ColumnDefinitions> {}
```
and impl:
```

public class MutinyColumnDefinitionsImpl implements MutinyColumnDefinitions
{
    private Publisher<? extends ColumnDefinitions> publisher;
    private final Uni<? extends ColumnDefinitions> uni;

    public MutinyColumnDefinitionsImpl(Publisher<? extends ColumnDefinitions> publisher)
    {
        this.publisher = publisher;
        this.uni = Wrappers.toUni(publisher);

    }

    @Override public void subscribe(Subscriber<? super ColumnDefinitions> subscriber)
    {
        publisher.subscribe(subscriber);
    }
```
but it would mean that we need to create 3 additional classes that basically do almost the same.